tasks/check_staged_filenames.js

```diff
@@ -0,0 +1,14 @@
+export default function (grunt) {
+  grunt.registerTask('checkStagedFilenames', function () {
+    grunt.task.requires('collectStagedFiles');
+
+    const invalid = grunt.config
+    .get('stagedFiles')
+    .filter(name => name.match(/[A-Z ]/))
+    .reduce((all, name) => `${all}  ${name}\n`, '');
+
+    if (invalid) {
+      grunt.log.error(`Filenames must use snake_case.\n${invalid}`);
```

Comment:
Should this be a `grunt.fail.warn` instead? As it is, this still lets you commit files with bad names.

![screenshot 2015-11-24 16 27 10](https://cloud.githubusercontent.com/assets/404731/11383886/889b7b50-92c8-11e5-8f6e-abfc4c27cd90.png)